### PR TITLE
Submit quizzes week03

### PR DESCRIPTION
**Pull request by mattfanz**

### CODE REVIEW

File: week0003/mattfanz/php/src/FinTech/AccountManager.php [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    }
+
+    /**
+     * @param $name
+     * @param $amount
+     * @return bool
+     */
+    public function addAccount($name, $amount)
+    {
+        if (ctype_alpha($name) && !$this->findOrFail($name) && is_numeric($amount) && $amount >= 0) {
```
Comment (vietlq) [inline week0003/mattfanz/php/src/FinTech/AccountManager.php]: You can't create an empty account ;) The amount is always positive!

File: week0003/mattfanz/php/src/FinTech/Action.php [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+     * @var AccountManager
+     */
+    protected $accountManager;
+
+    /**
+     * @var string
+     */
+    protected $output;
+
+    const ACTION_ADD_CLIENT = 'A';
```
Comment (vietlq) [inline week0003/mattfanz/php/src/FinTech/Action.php]: Good!

File: week0003/mattfanz/php/src/index.php [hunk 3]
```diff
@@ -0,0 +1,16 @@
+<?php
+require_once('autoload.php');
+use FinTech\AccountManager;
+use FinTech\Action;
+
+$accountManager = new AccountManager();
+
+while ($input = fgets(STDIN)) {
+    sscanf($input, "%s %s %s", $action, $name, $amount);
+    if ($action == 'E') {
```
Comment (vietlq) [inline week0003/mattfanz/php/src/index.php]: Good thing but we have no action 'E' in the requirement so remove it for now.

File: week0003/mattfanz/php/src/test_case_01.output [hunk 4]
```diff
@@ -0,0 +1,12 @@
+True
```
Comment (vietlq) [inline week0003/mattfanz/php/src/test_case_01.output]: You don't need to copy this file

File: week0003/mattfanz/php/src/index.php [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+require_once('autoload.php');
+use FinTech\AccountManager;
+use FinTech\Action;
+
+$accountManager = new AccountManager();
+
+while ($input = fgets(STDIN)) {
+    sscanf($input, "%s %s %s", $action, $name, $amount);
+    $newAction = new Action($action, $name, $amount, $accountManager);
+    $output = $newAction->getOutput();
```
Comment (vietlq) [inline week0003/mattfanz/php/src/index.php]: `performAction`, `execute` is more relevant than `getOutput`.

File: week0003/mattfanz/php/src/FinTech/Action.php [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+    const ACTION_DEPOSIT_MONEY = 'D';
+    const ACTION_WITHDRAW_MONEY = 'W';
+
+    /**
+     * Action constructor.
+     * @param $action
```
Comment (vietlq) [inline week0003/mattfanz/php/src/FinTech/Action.php]: I love documentation like this ;)

Comment (mattfanz) [inline week0003/mattfanz/php/src/FinTech/Action.php]: yes, thanks you. Just try to make it look good

File: week0003/mattfanz/php/src/FinTech/Action.php [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+     * @param AccountManager $accountManager
+     */
+    public function __construct($action, $name, $amount, AccountManager $accountManager)
+    {
+        $this->action = $action;
+        $this->name = $name;
+        $this->amount = $amount;
+        $this->accountManager = $accountManager;
+        
+        switch ($this->action) {
```
Comment (vietlq) [inline week0003/mattfanz/php/src/FinTech/Action.php]: This must be not in the constructor! What if you decide not to perform the action?

Move it to the performAction function.

**Comment by vietlq** [comment]
Ok now merge agains my master before I accept

**Comment by mattfanz** [comment]
can not merge.
`
git fetch upstream
git checkout master
git merge upstream/master
`

=> Already up-to-date.

**Pull request merged**